src/app/components/merge/MergeModal.js

```diff
@@ -0,0 +1,254 @@
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+import { c } from 'ttag';
+import { useApi, useEventManager, useModals, FormModal, ResetButton, PrimaryButton } from 'react-components';
+
+import { getContact, addContacts, deleteContacts } from 'proton-shared/lib/api/contacts';
+import { prepareContact as decrypt, bothUserKeys } from '../../helpers/decrypt';
+import { prepareContact as encrypt } from '../../helpers/encrypt';
+import { merge } from '../../helpers/merge';
+import { move } from 'proton-shared/lib/helpers/array';
+import { noop } from 'proton-shared/lib/helpers/function';
+import { OVERWRITE, CATEGORIES, SUCCESS_IMPORT_CODE } from '../../constants';
+
+import ContactDetails from './ContactDetails';
+import MergeContactPreview from './MergeContactPreview';
+import MergeModalContent from './MergeModalContent';
+import MergingModalContent from './MergingModalContent';
+
+const { OVERWRITE_CONTACT } = OVERWRITE;
+const { IGNORE } = CATEGORIES;
+
+// helper to re-order arrays inside a list of arrays
+const reOrderInGroup = (collection, groupIndex, { oldIndex, newIndex }) =>
```

Comment:
If it's a helper, feel free to extract it to an other file.

Reply:
Yeah, I didn't extract it because I saw no natural place where to do it. I created an `array` helper just with that function in the end